frontend/src/metabase-lib/lib/Dimension.ts

```diff
@@ -815,80 +818,123 @@ export class FieldDimension extends Dimension {
     return field;
   }
 
-  field(): Field {
+  _getIdentifierProp() {
+    return this.isIntegerFieldId() ? "id" : "name";
+  }
+
+  _getTrustedFieldCachedOnInstance() {
     if (
       this._fieldInstance &&
       this._fieldInstance._comesFromEndpoint === true
     ) {
       return this._fieldInstance;
     }
+  }
 
-    const question = this.query()?.question();
-    const lookupField = this.isIntegerFieldId() ? "id" : "name";
-    const fieldMetadata = question
-      ? _.findWhere(question.getResultMetadata(), {
-          [lookupField]: this.fieldIdOrName(),
-        })
-      : null;
+  _getFieldMetadataFromSavedQuestion() {
```

Comment:
all of these functions I'm adding that are appended with a `_` are fairly similar in that we are peaking into connected cards/queries/tables to glean information about the dimension itself. The fact that we need these functions is a failure of our abstractions and I think probably should not live on the `FieldDimension` itself. We should eventually figure out a way to handle this from the query/question/table.